src/controllers/simplified-account/settings/card-payments/google-pay.controller.js

```diff
@@ -1,37 +1,90 @@
+const paths = require('@root/paths')
 const { response } = require('@utils/response')
-const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const { formatSimplifiedAccountPathsFor, formatValidationErrors } = require('@utils/simplified-account/format')
 const { updateAllowGooglePay } = require('@services/card-payments.service')
-const paths = require('@root/paths')
-const validateOnOffToggle = require('@utils/simplified-account/validation/on-off-toggle')
+const { validationResult } = require('express-validator')
+const { WORLDPAY } = require('@models/constants/payment-providers')
+const { updateGooglePayMerchantId } = require('@services/worldpay-details.service')
+const { googlePaySchema, GOOGLE_PAY_TOGGLE_FIELD, GOOGLE_PAY_MERCHANT_ID_FIELD } = require('@utils/simplified-account/validation/google-pay.schema')
 
+/**
+ * @param {import('@utils/types/settings/settings-request').SettingsRequest} req
+ * @param {import('express').Response} res
+ */
 function get (req, res) {
-  const googlePay = req.account?.allowGooglePay
+  const account = req.account
+  const service = req.service
   response(req, res, 'simplified-account/settings/card-payments/google-pay', {
-    currentState: googlePay ? 'on' : 'off',
-    backLink: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.cardPayments.index, req.service.externalId, req.account.type)
+    currentState: account.allowGooglePay ? 'on' : 'off',
+    paymentProvider: account.paymentProvider,
```

Comment:
Should test for the value of this field in the google-pay.controller.test.js

Reply:
covered in a separate PR